it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/TaskListControllerISpec.scala

```diff
@@ -0,0 +1,33 @@
+package uk.gov.hmrc.agentsubscriptionfrontend.controllers
+import uk.gov.hmrc.agentsubscriptionfrontend.models.AgentSession
+import uk.gov.hmrc.agentsubscriptionfrontend.support.BaseISpec
+import uk.gov.hmrc.agentsubscriptionfrontend.support.SampleUser.subscribingAgentEnrolledForNonMTD
+
+class TaskListControllerISpec extends BaseISpec {
+  lazy val controller: TaskListController = app.injector.instanceOf[TaskListController]
+
+  "showTaskList (GET /task-list)" should {
+    behave like anAgentAffinityGroupOnlyEndpoint(controller.showTaskList(_))
+
+    "contain page titles and header content" in {
+      val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+      val result = await(controller.showTaskList(request))
+
+      result should containMessages(
+        "task-list.header",
+        "task-list.1.header",
+        "task-list.2.header",
+        "task-list.3.header",
+        "task-list.4.header")
+    }
+    "contain CONTINUE tag when a task has been completed" in {
+      implicit val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+      sessionStoreService.currentSession.agentSession = Some(AgentSession(identifyBusinessTaskComplete = true))
```

Comment:
we need to add a test where we have a session but no business details exists, then verify if we actually show the flag as true or false

Reply:
So the flag is set upon the submission of /confirm-business back to the track page, it's not based on the business details in the session